lib/hal/debug.h

```diff
@@ -23,13 +23,21 @@ extern "C"
  * Prints a message to whatever debug facilities might
  * be available.
  */
+__attribute__((deprecated("deprecated as of 05/2021", "Use SDL2 for generic text output")))
```

Comment:
Surprised you can have comma separated messages? GNU gcc man-pages say:

``` 
deprecated
deprecated (msg)
```

Also not sure if the message should contain the date like that.

Reply:
Unresolved, because the question about clang docs remain: is `replacement` meant as an auto-replacement hint that must be machine readable?